instrumentation.go

```diff
@@ -441,16 +401,6 @@ func WithEnv() InstrumentationOption {
 				c.globalImpl = boolVal
 			}
 		}
-		if l, ok := lookupEnv(envLogLevelKey); ok {
```

Comment:
If I understand this correctly, this change implies that `OTEL_LOG_LEVEL` is only being inspected when the instrumentation is launched from the CLI. When the instrumentation is being called as a library it won't be inspected? If that is the case I think we should change this behavior.

Reply:
That is correct. The `Instrumentation` no longer sets the minimum severity of the logger because it no longer constructs the logger. The `slog.Logger` is not something that is configured with this level it is something that the caller will need to setup.

Meaning the caller is given the opportunity to decide what ever way they want to configure this setting, or not.

Reply:
If the `WithLogger` is not used, I'd expect `OTEL_LOG_LEVEL` to still have an impact

Reply:
I've captured the many short-falls of our existing logging and examples of things I would like to achieve in https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/1104. This PR serves to address all of those.

I do not think we can force the `OTEL_LOG_LEVEL` on everyone and assume it will meet their needs. Accepting a `Logger` will still allow for this configuration, but will also allow for much more.

Reply:
If the user doesn't pass the logger, we should still use `OTEL_LOG_LEVEL` to setup our default.

Reply:
Updated. PTAL.